File: modules/wem-api/src/main/java/com/enonic/wem/api/content/page/text/TextComponentType.java
package com.enonic.wem.api.content.page.text;

import com.enonic.wem.api.content.page.PageComponentType;

public final class TextComponentType
    extends PageComponentType
{
    public final static TextComponentType INSTANCE = new TextComponentType();

    private static final TextComponentDataSerializer dataSerializer = new TextComponentDataSerializer();

    private TextComponentType()
    {
        super( "text", TextComponent.class );
    }

    @Override
    public TextComponentDataSerializer getDataSerializer()
    {
        return dataSerializer;
    }
}

File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/content/page/text/TextComponentType.ts
module api.content.page.text {

    import ComponentType = api.content.page.ComponentType;

    export class TextComponentType extends ComponentType {

        constructor() {
            super("text");
        }

        newComponentBuilder(): TextComponentBuilder {
            return new TextComponentBuilder();
        }
    }

    new TextComponentType();

}